### PR TITLE
Sept 23 upstream 1 (c9db002)

### PR DESCRIPTION
**Pull request by illusional**
This merge was not as smooth as I would've liked 😓

### CODE REVIEW

File: hail/python/hailtop/hailctl/dataproc/submit.py [hunk 1]
```diff
@@ -2,81 +2,67 @@
...
-            zipf.close()
+            with zipfile.ZipFile(tfile, 'w', zipfile.ZIP_DEFLATED) as zipf:
+                for hail_script_entry in pyfiles_list:
+                    if os.path.isfile(hail_script_entry) and _filter_pyfile(hail_script_entry):
+                        zipf.write(hail_script_entry, arcname=os.path.basename(hail_script_entry))
+                    else:
+                        for root, _, pyfiles_walk in os.walk(hail_script_entry):
+                            for pyfile in pyfiles_walk:
+                                path = os.path.join(root, pyfile)
+                                if os.path.isfile(path) and _filter_pyfile(path):
```
Comment (jmarshall) [inline hail/python/hailtop/hailctl/dataproc/submit.py:42]: Our previous used the `path` variable in `zip.write`'s arguments too, but I agree it's better this way — keep those lines the same as upstream.

Comment (illusional) [reply hail/python/hailtop/hailctl/dataproc/submit.py:42]: I manually rewrote this block to by a hybrid of above, part of the painful merge :sweat:
Doesn't look like we can feasibly revert back to upstream

Comment (jmarshall) [reply hail/python/hailtop/hailctl/dataproc/submit.py:42]: Yep, what I'm saying is I'm happy with it :+1:

**Comment by jmarshall** [comment]
LGTM.

In lieu of a better way to describe it, I've pushed an extra commit that reinstates two-blank-lines around our added `get_completed_batches_ordered_by_completed_time` function. Looks like _batch/batch/front_end/front_end.py_ was extremely not fun to merge! You can keep that extra commit, incorporate it, or ignore it as you wish.

**Review by jmarshall — APPROVED** [review]

**Pull request merged**